docs/source/cpp/env_vars.rst

```diff
@@ -85,6 +85,28 @@ that changing their value later will have an effect.
    ``libhdfs.dylib`` on macOS, ``libhdfs.so`` on other platforms).
    Alternatively, one can set :envvar:`HADOOP_HOME`.
 
+.. envvar:: ARROW_S3_LOG_LEVEL
```

Comment:
Let me know how you feel about the name here. I couldn't decide between giving it an ARROW prefix or an AWS one.

Reply:
I think that it's OK. It seems there is no common environment variable for log level used in AWS related tools.